feature/platform/fabric/otg_tests/sampled_backplane_capacity_counters_test/README.md

```diff
@@ -0,0 +1,90 @@
+# gNMI-1.18: gNMI subscribe with sample mode for backplane capacity counters
+
+## Summary
+WBB is required to support gNMI Subscribe with SAMPLE or ON_CHANGE mode for various counters.
+This test if to verify that DUT supports gNMI Subscribe with sample mode, updating
+the available backplane capacity counters correctly while forwarding traffic
+
+*   Get backplace capacity before any traffic is sent
+
+*   Send some traffic and wait for the sample to be collected
+
+*   Send more traffic and wait for the sample to be collected
+
+## Testbed type
+
+*   https://github.com/openconfig/featureprofiles/blob/main/topologies/atedut_2.testbed
+
+## Procedure
+
+### gNMI-1.18.1 [TODO: https://github.com/openconfig/featureprofiles/issues/2322]
+
+*   Connect DUT port-1 and 2 to ATE port-1 and 2 respectively
+
+    *   For MFF DUT ports 1 and 2 can be on different linecards
+
+*   Configure IPv4/IPv6 addresses on the  interfaces
+
+    *   Test with two sample intervals of 10 and 15 seconds
+ 
+    *   Using gNMI subscribe with "SAMPLE" mode
+
+    *   Initiate traffic:
+
+        *   Initiate traffic as per below threshold:
+    
+             Port number   | Interface1(line rate %)
+            -------------- | -----------------------
+            Port1          | 20%
+            Port2          | 20%
+
+     *   Validate we get capacity metrics every 10 and 15 seconds
+
+### gNMI-1.18.2 [TODO: https://github.com/openconfig/featureprofiles/issues/2323]
+
+*   Connect DUT port-1 and 2 to ATE port-1 and 2 respectively
+
+    *   For MFF DUT ports 1 and 2 can be on different linecards
+
+*   Configure IPv4/IPv6 addresses on the  interfaces
+
+    *   Using gNMI subscribe with "ON_CHANGE" mode
+
+    *   Initiate traffic:
+
+        *   Initiate traffic as per below threshold:
+
+             Port number   | Interface1(line rate %)
+            -------------- | -----------------------
+            Port1          | 20%
+            Port2          | 20%
+
+    *   Change traffic threshold to 50%
+
+    *   Again change the threshold to 70% after a few seconds
+
+    *   validate we get changed/updated capacity metrics
```

Comment:
which capacity?  Should both `consumed-capacity` and `total-operational-capacity` change?    Probably total-operational-capacity should not change.

Also, for a 'gauge' and counter type leafs  like  `consumed-capacity` which we expect continuous change, I think we should only require 'SAMPLE' mode.    

`total-operational-capacity` could be ON_CHANGE, and might be testable by disabling some fabric modules.

Reply:
```
/components/component/integrated-circuit/backplane-facing-capacity/state/

total
description: Total backplane-facing capacity that is available in the presence of no link failures or degradation.
units: bits per second

total-operational-capacity
description: Total backplane-facing capacity that is currently available based on the active links.
units: bits per second

consumed-capacity
description: Backplane-facing capacity that is consumed by front-panel ports that are connected to the integrated circuit and are operationally up.
units: bits per second

available-pct
description: Percentage of the total backplane-facing capacity that is currently available to the front panel ports taking into account failures and/or degradation within the system. In the case that there is more backplane-facing capacity available than the front-panel ports consume, this value may be greater than 100%.
```

ON_CHANGE makes more sense for all of these paths. These metrics care about operational/active ports and not the actual utilisation of the active ports. But we could also check if SAMPLE is supported for these paths.
To keep the test minimal functionality I will just check if we get sample or on_change metrics (shut one of the links to trigger a change)

Reply:
I also interpret `consumed-capacity` as a value that changes with whether ports are up or down.  However,    I took a quick look at  what some vendors implement, they do utilization.      As a network engineer also, I think reporting capacity used by ports that are up and down is not how the all the fabric interfaces that I am aware of actually work.    Capacity of linecard to fabric  is based on connected and active fabric lanes.  The consumption of that capacity is based purely on utilization.  Not by front panel ports being turned on and off.   

I think we should update the OC description for that leaf to reflect this and proceed with a test that is expecting to measure utilization of the fabric.  Thoughts?

Reply:
ref we can use for updating the OC model for `consumed-capacity`: 
Arista: https://www.arista.com/en/support/toi/eos-4-23-1f/14398-7500-series-fabric-capacity-monitor
Cisco: https://www.cisco.com/c/en/us/td/docs/ios_xr_sw/iosxr_r3-8/general/man_pages/crpkg38/ar38fabr.pdf. (doesn't directly support fabric capacity, just reports channels)
Juniper https://www.juniper.net/documentation/us/en/software/junos/chassis/topics/ref/command/show-chassis-fabric-fpcs.html#show-chassis-fabric-fpcs-command__d21040e392. Also only shows fabric/fpc channel status, not capacity